### PR TITLE
✨ Helper Proxy & single express-hbs instance

### PR DESCRIPTION
**Pull request by ErisDS**
Requires #8223 to be merged first (contains those commits already squashed)

This is the meat and veg after the foundations are laid in #8223.

I've moved all requires from helpers to be required via a Proxy object. See notes below.

With that done, I've moved to having a single `engine` file in the theme which creates a hbs instance we use  for everything else.

This allows me to unskip a test I wrote earlier, as the hbs engine is no longer clobbered by other tests. ~~Magic ✨~~ - wait, this isn't working again. Need to revisit that 😞 

There is also one extra require of express-hbs, also properly managed with `.create()` in the error handler. This refs Part 2 in #8221.

The remaining commits are some cleanup.

--- 

**Notes on the Proxy concept**

This is similar to the concept of a Proxy we have in server/apps/proxy.js. Long term the two proxies could either be merged or form the concept of an internal vs external (public vs private) proxy - one for what apps can use, one for what internal apps can use? Not sure yet but it's materialising...

The concept of a proxy is basically controlled dependency injection, where we are explicit about what a module is allowed to have access to. If we had access to more ES6 syntax there are lots of nice things we could use, e.g. `const {SafeString, url} = require('./proxy'};`. And maybe even [Proxy](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Proxy) itself - definitely useful for enforcing permissions over the top of libraries for Apps, I think.

I've also been researching to find a simpler way to override the scope that a module has access to than the sandbox we have in apps. I.e. to be able to enforce that proxy is available, and nothing else, not just for helpers, but in a nicer way for apps as well so we can be more confident about the safety of them going forward.

### CODE REVIEW

**Comment by ErisDS** [comment]
rebased 😁

File: core/server/helpers/author.js [hunk 1]
```diff
@@ -10,32 +10,31 @@
...
+    SafeString = proxy.SafeString,
+    helpers = proxy.hbs.handlebars.helpers,
+    templates = proxy.templates,
+    url = proxy.url;
 
-author = function (options) {
+module.exports = function author(options) {
     if (options.fn) {
-        return hbs.handlebars.helpers.with.call(this, this.author, options);
+        return helpers.with.call(this, this.author, options);
```
Comment (kirrg001) [inline core/server/helpers/author.js]: The only feedback i can give is, that i find it a little bit less readable.
`helpers.with` vs `hbs.handlebars.helpers.with`

File: core/server/helpers/ghost_head.js [hunk 2]
```diff
@@ -6,19 +6,22 @@
...
-    filters = require('../filters'),
-    assetHelper = require('./asset'),
-    config = require('../config'),
     Promise = require('bluebird'),
-    labs = require('../utils/labs'),
-    utils = require('../utils'),
-    api = require('../api'),
-    settingsCache = require('../settings/cache');
+
+    // Aarrgh what to do with this?!
```
Comment (kirrg001) [inline core/server/helpers/ghost_head.js]: Wanne leave this comment as is?

**Comment by kirrg001** [comment]
> The concept of a proxy is basically controlled dependency injection, where we are explicit about what a module is allowed to have access to. If we had access to more ES6 syntax there are lots of nice things we could use, e.g. const {SafeString, url} = require('./proxy'};. And maybe even Proxy itself - definitely useful for enforcing permissions over the top of libraries for Apps, I thin

👍

**Comment by ErisDS** [comment]
Updated 👍

**Review by kirrg001 — APPROVED** [review]
LTGM!

**Pull request merged**